### ISSUE TITLE
Use the Kotlin Coroutines BOM

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects/spring-boot#16555]
When available (should be before Boot 2.2 GA), see https://youtrack.jetbrains.com/issue/KT-30042.

**Comment by snicoll** [comment]
Thanks for the nudge. If there is a dedicated bom for this, it must only include coroutines and no other artifacts (as per our policy of including BOMs here).

**Comment by sdeleuze** [comment]
Yes that the plan

**Comment by mbhave** [comment]
Blocked on [this](https://github.com/Kotlin/kotlinx.coroutines/issues/1110).

**Comment by sdeleuze** [comment]
The Coroutines BOM [is now available](https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-bom/1.3.0-RC/kotlinx-coroutines-bom-1.3.0-RC.pom), could you ship it with Spring Boot 2.2.0.M5 ?

**Comment by snicoll** [comment]
@sdeleuze can you please upgrade the framework build to use it?

**Comment by sdeleuze** [comment]
@snicoll Yes I am trying to do so, but I am currently facing some issues related to broken classpath while my tests with the preview version were green, I will comment here asap it is ok.

**Comment by sdeleuze** [comment]
@snicoll Done https://github.com/spring-projects/spring-framework/commit/c006a83584052fdd48a5ec4722d327ad6356aa94

**Comment by philwebb** [comment]
Closing in favor of PR #17701
